### PR TITLE
add property and env var to disable connection creation

### PR DESCRIPTION
**Pull request by pwestling**

### CODE REVIEW

File: src/java/com/rapleaf/jack/BaseDatabaseConnection.java [hunk 1]
```diff
@@ -8,20 +8,34 @@
...
+import com.google.common.base.Optional;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public abstract class BaseDatabaseConnection implements Serializable, Closeable {
 
   private static Logger LOG = LoggerFactory.getLogger(BaseDatabaseConnection.class);
 
+  public static String DISABLE_PROPERTY = "jack.db.disallow_connections";
+  public static String DISABLE_ENV_VAR = "jack.db.disallow_connections";
```
Comment (bpodgursky) [inline src/java/com/rapleaf/jack/BaseDatabaseConnection.java]: it feels like JACK_DB_DISALLOW_CONNECTIONS would be more conventional for the environment variable.

File: src/java/com/rapleaf/jack/BaseDatabaseConnection.java [hunk 2]
```diff
@@ -165,4 +181,21 @@ public void close() throws IOException {
       throw new IOException(e);
     }
   }
+
+  private boolean disabled() {
```
Comment (bpodgursky) [inline src/java/com/rapleaf/jack/BaseDatabaseConnection.java]: not to hate on Optional, but I feel like boolean might be more straighforward here

````
boolean disabledBySystemProperty = false;
      try {
        disabledBySystemProperty = Boolean.parseBoolean(System.getProperty(DISABLE_PROPERTY));
       } catch (IllegalArgumentException e) {
         LOG.error("blah blah ignoring", e);
       }

      return disabledBySystemProperty || disabledByEnvVar;
````

Comment (bpodgursky) [inline src/java/com/rapleaf/jack/BaseDatabaseConnection.java]: also nbd but I'd probably call it isDisabled for clarity

**Comment by bpodgursky** [comment]
few comments, but looks good

File: src/java/com/rapleaf/jack/BaseDatabaseConnection.java (hunk 1 above)
Comment (pwestling) [inline src/java/com/rapleaf/jack/BaseDatabaseConnection.java]: fuck, copy/paste error

File: src/java/com/rapleaf/jack/BaseDatabaseConnection.java (hunk 2 above)
Comment (pwestling) [inline src/java/com/rapleaf/jack/BaseDatabaseConnection.java]: fair enough. Habits.

File: src/java/com/rapleaf/jack/BaseDatabaseConnection.java [hunk 3]
```diff
@@ -8,20 +8,34 @@
 import java.sql.SQLException;
 import java.sql.Statement;
 
+import com.google.common.base.Optional;
```
Comment (seancarr) [inline src/java/com/rapleaf/jack/BaseDatabaseConnection.java]: remove

File: src/java/com/rapleaf/jack/BaseDatabaseConnection.java [hunk 4]
```diff
@@ -8,20 +8,34 @@
...
   /**
    * Get a Connection to a database.
    * If there is no connection, create a new one.
    */
-  public abstract Connection getConnection();
+  public Connection getConnection() {
+    if (!isDisabled()) {
+      return getConnectionInternal();
+    } else {
+      throw new RuntimeException("Tried to instantiate a connection even though connection have been isDisabled in this environment");
```
Comment (sidoh) [inline src/java/com/rapleaf/jack/BaseDatabaseConnection.java]: > have been isDisabled in this environment

hehe, intellij. few places below this in comments where this happened too. not that it really matters :)

Comment (pwestling) [inline src/java/com/rapleaf/jack/BaseDatabaseConnection.java]: dammit intellij

**Pull request merged**